docs/src/modules/components/demos/data-grid/PTOCalendar/PTOCalendar.tsx

```diff
@@ -0,0 +1,697 @@
+          component="img"
+          src={`https://flagcdn.com/w40/${ptoData[params.row.employee].nationality.toLowerCase()}.png`}
+          alt={`${ptoData[params.row.employee].nationality} flag`}
+          sx={{
+            width: 18,
+            height: 18,
+            margin: '1px',
+            borderRadius: '50%',
+          }}
+        />
+        {showLabel && 'Holiday'}
+      </>
+    );
+  }
+  if (isFirstVisibleDayOfPTO && showPTO) {
+    return (
+      <>
+        <BeachAccessOutlined fontSize="small" />
+        {showLabel && 'Vacation'}
+      </>
+    );
+  }
+  if (isFirstDayOfSick && showSick) {
+    return (
+      <>
+        <DeviceThermostatOutlined fontSize="small" />
+        {showLabel && 'Sick leave'}
+      </>
+    );
+  }
+  if (isBirthday && cellData.show) {
+    return (
+      <>
+        <Cake fontSize="small" />
+        {showLabel && 'Birthday'}
+      </>
+    );
+  }
+  return null;
+}
+
+function PTOCalendar() {
+  const theme = useTheme();
+  const calendarState = useCalendarState();
+  const { currentDate, activeFilters, density } = calendarState;
+
+  const [holidays, setHolidays] = React.useState<HolidayData>({});
+  const ptoData = usePTOData(currentDate);
+
+  const monthStart = useMemo(() => startOfMonth(currentDate), [currentDate]);
+  const monthEnd = useMemo(() => endOfMonth(currentDate), [currentDate]);
+  const daysToShow = useMemo(
+    () => eachDayOfInterval({ start: monthStart, end: monthEnd }),
+    [monthStart, monthEnd],
+  );
+
+  useEffect(() => {
+    const fetchHolidays = async () => {
+      try {
+        const year = currentDate.getFullYear();
+        const countries = Array.from(
+          new Set(Object.values(ptoData).map((data) => data.nationality)),
+        );
+        const holidayData = await getHolidaysForCountries(year, countries);
+        setHolidays(holidayData);
+      } catch (error) {
+        console.error('Failed to fetch holidays:', error);
+        // TODO: Add proper error handling UI
+      }
+    };
+
+    fetchHolidays();
+  }, [currentDate, ptoData]);
+
+  const employeesOutOfOffice = useMemo(() => {
+    const summary: { [key: string]: number } = {};
+
+    daysToShow.forEach((day) => {
+      const dateStr = format(day, 'yyyy-MM-dd');
+      const count = Object.values(ptoData).filter(
+        (data) => data.ptoDates.includes(dateStr) || data.sickDates.includes(dateStr),
+      ).length;
+      summary[dateStr] = count;
+    });
+
+    return summary;
+  }, [daysToShow, ptoData]);
+
+  const pinnedRow: GridPinnedRowsProp = React.useMemo(
+    () => ({
+      top: [
+        {
+          id: 'summary',
+          employee: 'Out of office:',
+          ...Object.fromEntries(
+            daysToShow.map((day) => {
+              const dateStr = format(day, 'yyyy-MM-dd');
+              return [dateStr, employeesOutOfOffice[dateStr]];
+            }),
+          ),
+        },
+      ],
+    }),
+    [daysToShow, employeesOutOfOffice],
+  );
```

Comment:
```suggestion
  const pinnedRow: GridPinnedRowsProp = React.useMemo(() => {
    const row: any = { id: 'summary', employee: 'Out of office:' };
    daysToShow.forEach((day) => {
      const dateStr = format(day, 'yyyy-MM-dd');
      row[dateStr] = employeesOutOfOffice[dateStr];
    });
    return { top: [row] };
  }, [daysToShow, employeesOutOfOffice]);
```
A little bit more readable